### DISCUSSION TITLE
pyart.map.grid_from_radars gives strange values near the radar

### DISCUSSION THREAD

**Discussion by johnnypeste** [ARM-DOE/pyart#1675] [General]
![porto_radar_bug](https://github.com/user-attachments/assets/e5e84fff-b5c2-4a54-a9fc-e0272e782533)

Hi,
I'm using py-art to extract sigmet (Vaisala) data from Porto-Portugal radar (1047 m above sea level)

The pseudo cappi of reflectivity at 1000 m altitude seems OK, sufficiently far from radar, but they form some kind of a strange consisting of nodata value near the radar, as you can see from the attached figure.

Anyone can explain such behaviour?

Regards

Joao

PS

The python code code is the following:

import pyart

import matplotlib.pyplot as plt
import numpy as np
import cartopy.crs as ccrs

filename="input.iris"

radar = pyart.io.read_sigmet(filename)

#display = pyart.graph.RadarMapDisplay(radar)

# Setting projection and ploting the first tilt.
projection = ccrs.LambertConformal(
    central_latitude=radar.latitude["data"][0],
    central_longitude=radar.longitude["data"][0],
)

z_grid_limits = (-47.,1002.)
y_grid_limits = (-299580.9,299580.9)
x_grid_limits = (-299580.9,299580.9)

grid_resolution = 998.

def compute_number_of_points(extent, resolution):
    return int((extent[1] - extent[0])/resolution)

z_grid_points = compute_number_of_points(z_grid_limits, grid_resolution)
x_grid_points = compute_number_of_points(x_grid_limits, grid_resolution)
y_grid_points = compute_number_of_points(y_grid_limits, grid_resolution)

grid = pyart.map.grid_from_radars(radar,
                                  grid_shape=(z_grid_points,
                                              y_grid_points,
                                              x_grid_points),
                                  grid_limits=(z_grid_limits,
                                               y_grid_limits,
                                               x_grid_limits),
                                 )

pyart.io.write_grid("output.nc", grid, format='NETCDF4', include_fields='reflectivity', write_proj_coord_sys=True, proj_coord_sys=None, arm_time_variables=False, arm_alt_lat_lon_variables=False, write_point_x_y_z=False, write_point_lon_lat_alt=False)

**Comment by scollis** [comment]
Looks like some sort of interaction between the beam spacing and radius of influence. What was the scanning pattern of the radar (sweep angles)? Are you doing a CAPI at 1000M ASL or AGL?   From: johnnypeste ***@***.***>Date: Monday, October 28, 2024 at 07:37To: ARM-DOE/pyart ***@***.***>Cc: Subscribed ***@***.***>Subject: [ARM-DOE/pyart] pyart.map.grid_from_radars gives strange values near the radar (Discussion #1675)porto_radar_bug.jpg (view on web)Hi,I'm using py-art to extract sigmet (Vaisala) data from Porto-Portugal radar (1047 m above sea level)The pseudo cappi of reflectivity at 1000 m altitude seems OK, sufficiently far from radar, but they form some kind of a strange consisting of nodata value near the radar, as you can see from the attached figure.Anyone can explain such behaviour?RegardsJoaoPSThe python code code is the following:import pyartimport matplotlib.pyplot as pltimport numpy as npimport cartopy.crs as ccrsfilename="input.iris"radar = pyart.io.read_sigmet(filename)#display = pyart.graph.RadarMapDisplay(radar)Setting projection and ploting the first tilt.projection = ccrs.LambertConformal(central_latitude=radar.latitude["data"][0],central_longitude=radar.longitude["data"][0],)z_grid_limits = (-47.,1002.)y_grid_limits = (-299580.9,299580.9)x_grid_limits = (-299580.9,299580.9)grid_resolution = 998.def compute_number_of_points(extent, resolution):return int((extent[1] - extent[0])/resolution)z_grid_points = compute_number_of_points(z_grid_limits, grid_resolution)x_grid_points = compute_number_of_points(x_grid_limits, grid_resolution)y_grid_points = compute_number_of_points(y_grid_limits, grid_resolution)grid = pyart.map.grid_from_radars(radar,grid_shape=(z_grid_points,y_grid_points,x_grid_points),grid_limits=(z_grid_limits,y_grid_limits,x_grid_limits),)pyart.io.write_grid("output.nc", grid, format='NETCDF4', include_fields='reflectivity', write_proj_coord_sys=True, proj_coord_sys=None, arm_time_variables=False, arm_alt_lat_lon_variables=False, write_point_x_y_z=False, write_point_lon_lat_alt=False)—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Reply by johnnypeste** [reply]
I want ASL (MSL=1000m) , so I did z=-47 (since the radar is at MSL=1047 m) as you can check in the code...

What was the scanning pattern of the radar (sweep angles)?

What is the parameter that you want me to check in raw data?

**Reply by johnnypeste** [reply]
radar.elevation['data'][0 to 359] = -0.08788086
radar.elevation['data'][360 to 719] = 0.06591797
radar.elevation['data'][720 to 1079] =0.9667968
radar.elevation['data'][1080 to 1439] =1.8676754
radar.elevation['data'][1440 to 1799] =2.7685542

**Comment by johnnypeste** [comment]
[PRT221213140528.zip](https://github.com/user-attachments/files/17543341/PRT221213140528.zip)
I'm sending the original Sigmet (Vaisala) Raw Data where I hope you can reproduce the strange cross pattern at radar site, when producing a CAPI Z

**Comment by johnnypeste** [comment]
I think I solved my problem, but I'm not sure what are the consequences:

So I changed the default  min_radius from 250 m to 1000 m, and the strange cross pattern disappeared.
The rest of the spatial pattern seems to be the same...

The code is now:

grid = pyart.map.grid_from_radars(radar,
                                  gridding_algo='map_gates_to_grid',
                                  roi_func="dist_beam",
                                  constant_roi=None,
                                  min_radius=1000.0, h_factor=(1.0, 1.0, 1.0), nb=1.0, bsp=1.0,
                                  grid_shape=(z_grid_points,
                                              y_grid_points,
                                              x_grid_points),
                                  grid_limits=(z_grid_limits,
                                               y_grid_limits,
                                               x_grid_limits),
                                 )

**Comment by scollis** [comment]
Yep.For a good overview of gridding see Trapp and Doswellhttps://journals.ametsoc.org/view/journals/atot/17/2/1520-0426_2000_017_0105_rdoa_2_0_co_2.xml  From: johnnypeste ***@***.***>Date: Monday, October 28, 2024 at 12:06To: ARM-DOE/pyart ***@***.***>Cc: Scott Collis ***@***.***>, Comment ***@***.***>Subject: Re: [ARM-DOE/pyart] pyart.map.grid_from_radars gives strange values near the radar (Discussion #1675)I think I solved my problem, but I'm not sure what are the consequences:So I changed the default min_radius from 250 m to 1000 m, and the strange cross pattern disappeared.The rest of the spatial pattern seems to be the same...The code is now:grid = pyart.map.grid_from_radars(radar,gridding_algo='map_gates_to_grid',roi_func="dist_beam",constant_roi=None,min_radius=1000.0, h_factor=(1.0, 1.0, 1.0), nb=1.0, bsp=1.0,grid_shape=(z_grid_points,y_grid_points,x_grid_points),grid_limits=(z_grid_limits,y_grid_limits,x_grid_limits),)—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Comment by johnnypeste** [comment]
Thanks a lot... It seems a rather complex article for me to explore it now... I will proceed as it is...
